src/librustc/ty/print/pretty.rs

```diff
@@ -988,16 +1014,33 @@ pub trait PrettyPrinter<'tcx>:
                 };
                 p!(print_value_path(instance.def_id(), instance.substs));
             }
-            // For zsts just print their type as their value gives no extra information
-            (Scalar::Raw { size: 0, .. }, _) => p!(print(ty)),
+            // For function type zsts just printing the type is enough
+            (Scalar::Raw { size: 0, .. }, ty::FnDef(..)) => p!(print(ty)),
```

Comment:
Probably worth using `p!(print_value_path(def_id, substs))`.
Would perhaps also allow removing whatever special-casing `TerminatorKind::Call` relies on?

Reply:
There's no special casing. It just prints the `func` `Operand` directly: `write!(fmt, "{:?}(", func)?;`

Reply:
Then `Operand` or `Constant` has special-casing?

Reply:
There's none there either.

Reply:
This is where function call terminators print their callee operand: https://github.com/rust-lang/rust/blob/master/src/librustc/mir/mod.rs#L1488

Which just forwards to debug printing the constant

https://github.com/rust-lang/rust/blob/master/src/librustc/mir/mod.rs#L2018

Which just calls `pretty_print_const`. So the previous code must have somehow handled this nicely without any special code?

Reply:
Found it: https://github.com/rust-lang/rust/blob/e5e8ba4edc435c9f87314b23a6c5d9c175bdf19c/src/librustc/ty/print/pretty.rs#L853

Can you try removing that line and making sure all test continue to pass?